jetty-server/src/main/java/org/eclipse/jetty/server/HttpOutput.java

```diff
@@ -1510,6 +1510,12 @@ public void run()
         }
     }
 
+    // For testing
+    ApiState getApiState()
+    {
+        return _apiState;
+    }
```

Comment:
I personally do not have a problem with exposing internals for testing reasons.

Especially in these kinds of deep classes where extending them is more complicated than it's worth.

Reply:
The state is provided in the toString output, so a test that checks for it as a substring is probably as robust as this and doesn't require an extra method.

Reply:
Okay, I got rid of the extra method and extracted the api state from `toString`.